### PR TITLE
feat: Added `latest_restorable_time` to ignore_changes

### PR DESCRIPTION
**Pull request by FieryCod**
Ignore `latest_restorable_time` property

## Description

```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the
last "terraform apply":

  # module.xxx.module.db.module.db_instance.aws_db_instance.this[0] has been changed
  ~ resource "aws_db_instance" "this" {
        id                                    = "xxx"
      ~ latest_restorable_time                = "2021-09-20T18:59:05Z" -> "2021-09-20T19:04:05Z"
        name                                  = "xx"
        tags                                  = {
            "Environment" = "staging"
            "Name"        = "xxxx"
            "Owner"       = "xxx"
        }
        # (57 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Unless you have made equivalent changes to your configuration, or ignored the
relevant attributes using ignore_changes, the following plan may include
actions to undo or respond to these changes.
```
## Motivation and Context

I would rather not see the outside change of `lastest_restorable_time` on every apply.

## Breaking Changes

None
## How Has This Been Tested?
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

**Comment by drmaciej** [comment]
Hi @FieryCod is there any chance you could fix this PR so it's in a mergeable state? I would really love to see it, this change on lastest_restorable_time is really annoying to me as well :)

**Comment by FieryCod** [comment]
@drmaciej done.

**Comment by drmaciej** [comment]
@FieryCod thanks much!
let's now hope the maintainers can review this soon 🤞

**Review by bryantbiggs — APPROVED** [review]
@antonbabenko I believe this one should be acceptable since its an attribute not an argument https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#latest_restorable_time

related to upstream AWS provider issue https://github.com/hashicorp/terraform-provider-aws/issues/19727

**Comment by antonbabenko** [comment]
I agree. `Objects have changed outside of Terraform` type of errors are rather painful ([20 open issues mentioning it](https://github.com/hashicorp/terraform-provider-aws/issues?q=%22Objects+changed+outside+of+Terraform%22)).

Let's fix the CI and it will be ready to merge.

**Review by korovkintaptap — APPROVED** [review]

**Comment by korovkintaptap** [comment]
can we please merge this ?

File: modules/db_instance/main.tf [hunk 1]
```diff
@@ -115,6 +115,12 @@ resource "aws_db_instance" "this" {
     delete = lookup(var.timeouts, "delete", null)
     update = lookup(var.timeouts, "update", null)
   }
+
```
Comment (antonbabenko) [inline modules/db_instance/main.tf:118]:
```suggestion

```

File: modules/db_instance/main.tf [hunk 2]
```diff
@@ -115,6 +115,11 @@ resource "aws_db_instance" "this" {
     delete = lookup(var.timeouts, "delete", null)
     update = lookup(var.timeouts, "update", null)
   }
+  lifecycle {
```
Comment (antonbabenko) [inline modules/db_instance/main.tf:118]:
```suggestion

  lifecycle {
```

**Comment by antonbabenko** [comment]
This PR is included in version 3.5.0 :tada:

**Comment by korovkintaptap** [comment]
awesome, unfortunately it doesn't work though 
i forked and tried the change, it doesn't change anything

**Comment by bryantbiggs** [comment]
> i forked and tried the change, it doesn't change anything

could you elaborate? the lifecycle ignore is intended to ignore any future changes so if that is happening then it does work. if `latest_restorable_time` is still showing up in diffs, then its not working

just a few more words/details makes a world of a difference 😉

**Comment by danielcompton** [comment]
I have tested this and found that this PR doesn't have the desired effect. 

I updated to 3.5.0 and ran `terraform apply -refresh-only`. After waiting 10 minutes, running `terraform plan` shows drift again from `latest_restorable_time`.

I don't think that `ignore_changes` works the way that we'd want it to for this case, see https://github.com/hashicorp/terraform/issues/28803#issuecomment-951209299 for more info on this.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.0.0 :tada:

**Pull request merged**